scripts/pylib/twister/twisterlib/statuses.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Status classes to be used instead of str statuses.
+"""
+
+from enum import Enum
+
+
+class TestInstanceStatus(str, Enum):
+    NONE = None  # to preserve old functionality
+    ERROR = 'error'
+    FAIL = 'failed'
+    FILTER = 'filtered'
+    FLASH = 'flash_error'
```

Comment:
that should not be exposed as status, needs to clean this up

Reply:
It is a status currently, thus should be categorised analogously to every other status. As I assume we would like to excise it from Zephyr, we ought to do it in the next step of our status overhaul. That next step will also be easier with every use of such status highlighted.

Reply:
please remove it now, as it is a leftover and does not propagte as status in current reports, this should translate to simple error with a reason of flash error.  see here https://github.com/zephyrproject-rtos/zephyr/pull/61414/commits/2d9384d50aa7b265d21912296c5137712c09e2e2 and here https://github.com/zephyrproject-rtos/zephyr/pull/61414/commits/2de29afdcfc60a23d54fbd1980c1b2873b408811

Reply:
Removed.